apps/livreGUI/animationController/AnimationController.cpp

```diff
@@ -99,13 +99,13 @@ struct AnimationController::Impl
         _ui.btnPlay->setText( enable ? "Pause" : "Play" );
     }
 
-    void onNewFrameReceived( zeq::hbp::data::Frame frame )
+    void onNewFrameReceived()
     {
         _connected = true;
 
         const int32_t int32Max = std::numeric_limits<int32_t>::max();
-        const int32_t startFrame = std::min( (int32_t)frame.start, int32Max );
-        const int32_t endFrame = std::min( (int32_t)frame.end, int32Max );
+        const int32_t startFrame = std::min( (int32_t)_frame.getStart(), int32Max );
+        const int32_t endFrame = std::min( (int32_t)_frame.getEnd(), int32Max );
```

Comment:
Thinking about it, frame should have the same "logic layer" as zerobuf::data::Progress to enforce these constraints.